### ISSUE TITLE
Optional permissions for commands

### ISSUE THREAD

**Issue by tech6hutch** [dirigeants/klasa#498]
### Describe your proposal
Commands do not currently have any way to specify permissions they _use_, but do not _need_. I propose that an `optionalPermissions` option be added.

### Use-Cases for your proposal
I have a command that uses the MANAGE_MESSAGES permission. It's not required to run the command, but it's something that it uses, if it's available to the bot. However, when the invite command is used, the generated invite does not include this optional permission (as it has no way to know its wanted).

### Expected and actual behavior
Expected: any permission the bot uses, it asks for in its invite, and the inviter can choose whether to grant it or not.
Actual: I'd have to transfer the invite command or make a hidden command with the missing perms as its `requiredPermissions`, in order for the bot's invite to properly reflect what it uses.

**Comment by tech6hutch** [comment]
I'd rather specify permissions per-command (in the actual command that uses a permission), but thank you for that. I suppose I could extend `client.invite` to get that behavior without too much trouble.

**Comment by bdistin** [comment]
Yeah, I am against making Command any more complex than it is, especially if it doesn't do anything to benefit all commands, but rather just something for the invite command.
